javascript/algorithms/MergeSort.js

```diff
@@ -0,0 +1,36 @@
+var a = [149, 520, 50, 76, 270, 684, 128, 584, 290];
+ 
+function merge(left, right)
+{
+    var result = [];
+ 
+    while (left.length && right.length) {
+        if (left[0] <= right[0]) {
+            result.push(left.shift());
+        } else {
+            result.push(right.shift());
+        }
+    }
+ 
+    while (left.length)
+        result.push(left.shift());
+ 
+    while (right.length)
+        result.push(right.shift());
+ 
+    return result;
+}
+ 
+function mergeSort(arr)
```

Comment:
Check the Code Again

Reply:
I already checked. Did you know what is actually this function for?